File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        if(arr.size()==0)return 0;
        int cnt=0,mx=0;
        for(int i=0;i<arr.size();i++){
            mx = max(arr[i],mx);
            if(mx==i)cnt++;
        }
        return cnt;
    }
};

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        if len(arr)==0:
            return 0
        
        maxm=0
        cnt = 0
        for i in range(len(arr)):
            maxm = max(arr[i],maxm)
            if(maxm==i):
               cnt+=1
        return cnt       